ethereum/ewm/BREthereumClient.h

```diff
@@ -291,42 +291,46 @@ extern "C" {
 
     typedef struct {
         BREthereumWalletEventType type;
+        BREthereumStatus status;
         union {
             struct {
                 BREthereumCookie cookie;
                 BREthereumGas gasEstimate;
                 BREthereumGasPrice gasPrice;
             } feeEstimate;
         } u;
+        char errorDescription[16];
```

Comment:
Is the status code not enough? What is captured here?

Reply:
That `errorDescription` is filled in a single case where a transfer has failed.  It will have a string like 'gas limit exceeded' or 'gas price too low'.    Maybe the string could be avoided if `BREthereumStatus` listed up the failure reasons - which actually it might be able to do:
```
typedef enum {
    SUCCESS,
    // ...

    // Transfer
    ERROR_TRANSACTION_HASH_MISMATCH,
    ERROR_TRANSACTION_SUBMISSION,

    // ...
} BREthereumStatus;

Reply:
Yeah. For C, I"m certainly a fan of the single, massive error code type. It's not great but I haven't seen a better pattern in C.

Reply:
VxWorks, I think IOS, have a concept of encoding a pair of {<module>, <module-error>} into an integer as an error code.  module is a global namespace (or could be declared/assigned dynamically).